onnxruntime/core/graph/graph.cc

```diff
@@ -2088,7 +2166,7 @@ void Graph::AddInitializedTensor(const TensorProto& tensor) {
   *(tensor_added) = tensor;
   name_to_initial_tensor_[tensor.name()] = tensor_added;
 
-  if (!GraphLoadedFromModelFile(graph_proto_) && GetNodeArg(tensor.name()) == nullptr) {
+  if (!is_loaded_from_model_file_ && GetNodeArg(tensor.name()) == nullptr) {
     // make sure there is a NodeArg for the initializer as SetGraphInputsOutputs may add it to the graph inputs.
```

Comment:
Probably don't want this first condition any more as we're allowing adding nodes/inputs to a model loaded from proto so the need to create a node arg may exist.

Reply:
tried to remove the `!is_loaded_from_model_file` but this test started failing: GraphTransformationTests.EmbedLayerNormFusionFormat5

---
In reply to: [405174439](https://github.com/microsoft/onnxruntime/pull/3446#discussion_r405174439) [](ancestors = 405174439)

Reply:
there's a call to graph_utils::AddInitializer() which calls Graph::AddInitializedTensor() prior to calling Graph::GetOrCreateNodeArg(). The TypeProto* passed to the last GetOrCreateNodeArg() contains shape information, but that at that point the NodeArg already exists. So it doesn't get the shape set.

---
In reply to: [405762855](https://github.com/microsoft/onnxruntime/pull/3446#discussion_r405762855) [](ancestors = 405762855,405174439)

Reply:
i'll leave this as is for now to minimize changes to Graph

---
In reply to: [405764968](https://github.com/microsoft/onnxruntime/pull/3446#discussion_r405764968) [](ancestors = 405764968,405762855,405174439)